todomvc-app-react/src/sam/TodoApp.js

```diff
@@ -0,0 +1,218 @@
+import React, { useState } from 'react';
+import '../../node_modules/todomvc-common/base.css'
+import '../../node_modules/todomvc-common/base.css'
+
+const clone = (state) => {
```

Comment:
This one could use a comment. Very hard to read. No idea what it does.

Reply:
There is a recursive relationship in the state (.parent), do you know if I still need to clone the state for React to work properly? I tried without cloning and it processes the first event/rendering correctly, then there is no more rendering. I thought the hooks were solving that problem.